RealEstate/apps/api/serializers.py

```diff
@@ -14,4 +15,85 @@ def validate(self, attrs):
         if not homebuyer:
             msg = _('Only home buyers are allowed to use this functionality.')
             raise serializers.ValidationError(msg)
-        return user
\ No newline at end of file
+        return user
+
+class APIHouseSerializer(serializers.ModelSerializer):
+
+    def create(self, validated_data):
+        house = House(**validated_data)
+        house.save()
+        return house
+
+    class Meta:
+        model = House
+        field = ('nickname', 'address')
+
+class APICategoryWeightSerializer(serializers.ModelSerializer):
+
+    def update(self, instance, validated_data):
+        instance.weight = validated_data.get('weight', instance.weight)
+        instance.save()
+        return instance
+
+    class Meta:
+        model = CategoryWeight
+
+class APIGradeSerializer(serializers.ModelSerializer):
+
+    def update(self, instance, validated_data):
+        instance.score = validated_data.get('score', instance.score)
+        instance.update()
+        return instance
+
+    class Meta:
+        model = Grade
+        fields = ('score', 'homebuyer', 'house', 'category')
+
+class APIHouseParamSerializer(serializers.Serializer):
```

Comment:
Actually we don't need to because our API doesn't allow adding new instance. Since the bug related to "update" has not been resolved in this version, I left this part here and see if the new version will release shortly.


Friday, July 10, 2015, 11:28 PM -0700 from Souriya Khaosanga  <notifications@github.com>:
>In  RealEstate/apps/api/serializers.py :
>> +
> +    class Meta:
> +        model = CategoryWeight
> +
> +class APIGradeSerializer(serializers.ModelSerializer):
> +
> +    def update(self, instance, validated_data):
> +        instance.score = validated_data.get('score', instance.score)
> +        instance.update()
> +        return instance
> +
> +    class Meta:
> +        model = Grade
> +        fields = ('score', 'homebuyer', 'house', 'category')
> +
> +class APIHouseParamSerializer(serializers.Serializer):
>This class should implement all abstract methods
>def create(self, validated_data):
>#implement
>def update(self, instance, validated_data):
>#implement
>—
>Reply to this email directly or  view it on GitHub .